### DISCUSSION TITLE
Why I Cannot Access Secrets From Action?

### DISCUSSION THREAD

**Discussion by rusith** [community/community#26494] [Actions]
<p>I have posted the below question in StackOverflow because this forum doesn’t allow me to post images.</p>
<p><a href="https://stackoverflow.com/questions/61308519/why-the-action-cannot-access-secrets" rel="nofollow noopener">https://stackoverflow.com/questions/61308519/why-the-action-cannot-access-secrets</a></p>
<p>Please be kind enough to check it out and help me if possible.</p>
<p>Thanks in advance!</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/rusith">@rusith</a> ,</p>
<p>This is by designed, as <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#about-encrypted-secrets" rel="nofollow noopener">doc</a> mentioned below:</p>
<p><strong>Warning:</strong><span> GitHub automatically redacts secrets printed to the log, but you should avoid printing secrets to the log intentionally.</span></p>
<p>Please check the typical usage below:</p>
<pre><code>steps:
  - name: Hello world action
    with: # Set the secret as an input
      super_secret: ${{ secrets.SuperSecret }}
    env: # Or as an environment variable
      super_secret: ${{ secrets.SuperSecret }}
</code></pre>
<p>Thanks.</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/rusith">@rusith</a> ,</p>
<p>I notice your reply on the refered SO ticket, hence it’s resolved?</p>
<p>This problem has occurred because a misunderstanding of mine which is “The Secret Values Should Show in The Logs if They are Passed to the Action”.</p>
<p>I am combining the answers of<span> </span><a href="https://stackoverflow.com/a/61313152/4919731" target="_blank" rel="nofollow noopener">Ben Winding</a><span> </span>and<span> </span><a href="https://stackoverflow.com/users/8705432/bk2204" target="_blank" rel="nofollow noopener">bk2204<span> </span></a>to make it clear.</p>
<p><strong>Secret values are scrubbed in action logs</strong>. Don’t expect to see the actual values in the action log. Getting the scrubbed text means the value has been passed to the action. you can use the value but you cant see them in the logs. Check<span> </span><a href="https://stackoverflow.com/a/61313152/4919731" target="_blank" rel="nofollow noopener">Ben’s Answer</a><span> </span>for How you can see the values even though it is not recommended.</p>

**Comment by rusith** [comment]
<p>Hi <a class="mention" href="/weide-zhou">@weide-zhou</a> </p>
<p>Yes, It is resolved. thanks for the reply .</p>
